packages/google_adsense/lib/src/ad_unit_widget_web.dart

```diff
@@ -0,0 +1,201 @@
+    _insElement
+      ..className = 'adsbygoogle'
+      ..style.display = 'block';
+    if (cssText != null && cssText.isNotEmpty) {
+      _insElement.style.cssText = cssText;
+    }
+    final Map<String, String> dataAttrs =
+        Map<String, String>.of(<String, String>{
+      AdUnitParams.AD_CLIENT: 'ca-pub-$_adClient',
+      'adSlot': _adSlot,
+    });
+    if (_isAdTest) {
+      dataAttrs.addAll(<String, String>{_AD_TEST_KEY: 'on'});
+    }
+    for (final String key in dataAttrs.keys) {
+      _insElement.dataset
+          .setProperty(key as JSString, dataAttrs[key]! as JSString);
+    }
+    if (_additionalParams.isNotEmpty) {
+      for (final String key in _additionalParams.keys) {
+        _insElement.dataset.setProperty(
+            key as JSString, _additionalParams[key].toString() as JSString);
+      }
+    }
+  }
+
+  static const String _AD_TEST_KEY = 'adtest';
+
+  @override
+  String get adClient => _adClient;
+  final String _adClient;
+
+  @override
+  String get adSlot => _adSlot;
+  final String _adSlot;
+
+  @override
+  bool get isAdTest => _isAdTest;
+  final bool _isAdTest;
+
+  @override
+  Map<String, dynamic> get additionalParams => _additionalParams;
+  final Map<String, dynamic> _additionalParams;
+
+  final web.HTMLElement _insElement =
+      web.document.createElement('ins') as web.HTMLElement;
+
+  @override
+  State<AdUnitWidgetWeb> createState() => _AdUnitWidgetWebState();
+}
+
+class _AdUnitWidgetWebState extends State<AdUnitWidgetWeb>
+    with AutomaticKeepAliveClientMixin {
+  static int adUnitCounter = 0;
+  double adHeight = 1.0;
+  late web.HTMLElement adUnitDiv;
+
+  @override
+  bool get wantKeepAlive => true;
+
+  @override
+  Widget build(BuildContext context) {
+    super.build(context);
+    return SizedBox(
+      height: adHeight,
+      child: HtmlElementView.fromTagName(
+          tagName: 'div', onElementCreated: onElementCreated),
+    );
+  }
+
+  void onElementCreated(Object element) {
+    adUnitDiv = element as web.HTMLElement;
+    log('onElementCreated: $adUnitDiv with style height=${element.offsetHeight} and width=${element.offsetWidth}');
+    adUnitDiv
+      ..id = 'adUnit${adUnitCounter++}'
+      ..style.height = 'min-content'
+      ..style.textAlign = 'center';
+    // Adding ins inside of the adUnit
+    adUnitDiv.append(widget._insElement);
+
+    // TODO(sokoloff06): Make shared
+    // Using Resize observer to detect element attached to DOM
+    web.ResizeObserver((JSArray<web.ResizeObserverEntry> entries,
+            web.ResizeObserver observer) {
+      for (final web.ResizeObserverEntry entry in entries.toDart) {
+        final web.Element target = entry.target;
+        if (target.isConnected) {
+          // First time resized since attached to DOM -> attachment callback from Flutter docs by David
+          if (!(target as web.HTMLElement)
+              .dataset
+              .getProperty('attached' as JSString)
```

Comment:
Why do we need this `'attached'` flag? In case the widget rebuilds, we'll get a brand new `element`, so I don't think the `target` will ever repeat?